src/web/rustdoc.rs

```diff
@@ -260,7 +278,17 @@ pub fn badge_handler(req: &mut Request) -> IronResult<Response> {
                 }
             }
         }
-        None => {
+        MatchVersion::Semver(version) => {
+            let url = ctry!(Url::parse(&format!("{}://{}:{}/{}/badge.svg?version={}",
```

Comment:
I'm not actually super sure if it's necessary to redirect the badge? Since the version is added by a query string, i'm not sure if it's actually cached in the first place. On the other hand, since the bare URL is meant to serve the latest version, there's a chance that the path may be cached already.

Reply:
[Some cursory reading](https://stackoverflow.com/questions/9692665/cache-busting-via-params) tells me that caches treat the same query string as the same URL for the purposes of caching, so it's worth redirecting so that they load the proper badge. (Whether we want to alter the cache time on badges so that they're always up to date is another matter, though...)